OpenEdXMobile/src/main/java/org/edx/mobile/util/LocaleUtils.java

```diff
@@ -58,6 +58,30 @@ public static String getLanguageNameFromCode(@NonNull String languageCode) throw
         }
     }
 
+    /**
+     * Provides the list of languages fetched from the system.
+     *
+     * @return A list of {@link FormOption} containing language names and their codes.
+     */
+    public static List<FormOption> getLanguages() {
+        final List<FormOption> languages = new ArrayList<>();
+        for (String languageCode : Locale.getISOLanguages()) {
+            try {
+                final String languageName = getLanguageNameFromCode(languageCode);
+                /*
+                Ignore language codes that don't translate to a proper language name.
+                e.g. language names like aa, ace, ada, ady, ae etc.
+                */
+                if (Character.isUpperCase(languageName.charAt(0)) && languageName.length() > 3) {
```

Comment:
`Character.isUpperCase(languageName.charAt(0))` returns false in case of languges that not have upper/lower cases characters. So languages are not availabe for the particular languages.